### PR TITLE
portability: /bin/true -> /usr/bin/true

### PR DESCRIPTION
**Pull request by oliver-sanders**
This is a small change with no associated Issue.

* `true` is a shell builtin available everywhere.
* `/bin/true` is not available on all platforms.
* `/usr/bin/true` would appear to be present on most.

Options:

1) ~Change to `/usr/bin/true`.~ Not a valid option.
2) Change to `true`.

Happy to go either way.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Already covered by existing tests.
- [x] No change log entry required (why? e.g. invisible to users).
- [x] No documentation update required.

### CODE REVIEW

File: tests/shutdown/05-auto/suite.rc [hunk 1]
```diff
@@ -12,4 +12,4 @@
         P1D = foo
 [runtime]
     [[foo]]
-        script = /bin/true
+        script = /usr/bin/true
```
Comment (kinow) [inline tests/shutdown/05-auto/suite.rc:15]: Fails on Ubuntu for me.

```
kinow@ranma:~$ /usr/bin/true
bash: /usr/bin/true: No such file or directory
kinow@ranma:~$ which true
/bin/true
```

Travis has this same test failing too.

Comment (kinow) [reply tests/shutdown/05-auto/suite.rc:15]: (I like the branch name BTW :smile:)

**Comment by kinow** [comment]
Maybe we will need to use `/usr/bin/env true` ?

File: tests/shutdown/05-auto/suite.rc (hunk 1 above)
Comment (oliver-sanders) [reply tests/shutdown/05-auto/suite.rc:15]: Sounds like it'll have to be option (2) then.

**Comment by hjoliver** [comment]
We already use a lot of plain `true`. 
```
(venv) [oliverh@niwa-1007885 cylc-flow]$ git grep 'script = true' | wc -l
288
(venv) [oliverh@niwa-1007885 cylc-flow]$ git grep 'script = "true"' | wc -l                                                                                                                                        
27
 ```
So I reckon that's the best option.

**Comment by hjoliver** [comment]
(Particularly if, as @oliver-sanders, *`true` is a shell builtin available everywhere.*)

**Comment by hjoliver** [comment]
Ha, you bet me to it :grin:

File: cylc/flow/hostuserutil.py [hunk 2]
```diff
@@ -113,7 +113,7 @@ def _get_host_info(self, target=None):
         """Return the extended info of the current host."""
         if target not in self._host_exs:
             if target is None:
-                target = socket.getfqdn()
+                target = socket.gethostname()
             try:
```
Comment (hjoliver) [inline cylc/flow/hostuserutil.py:117]: What's the deal with this change?

**Comment by oliver-sanders** [comment]
Who would have thought true/false were non-standard!?

File: cylc/flow/hostuserutil.py (hunk 2 above)
Comment (oliver-sanders) [reply cylc/flow/hostuserutil.py:117]: Whoops.

BTW this is the change which makes Cylc work on latest Mac OS.

Comment (hjoliver) [reply cylc/flow/hostuserutil.py]: Ah, good to know.

**Comment by kinow** [comment]
One (non-flaky) test failed on Travis

```
./tests/restart/00-pre-initial.t                                (Wstat: 0 Tests: 4 Failed: 2)
  Failed tests:  3-4
```

Passed locally

```bash
(venv) kinow@ranma:~/Development/python/workspace/cylc-flow$ ./etc/bin/run-functional-tests.sh -v ./tests/restart/00-pre-initial.t

ok 1 - 00-pre-initial-validate
ok 2 - 00-pre-initial-run
ok 3 - 00-pre-initial-restart
ok 4 - final-state-contains-ok
ok    28972 ms ( 0.00 usr  0.00 sys +  8.62 cusr  0.94 csys =  9.56 CPU)
[09:15:18]
All tests successful.
Files=1, Tests=4, 30 wallclock secs ( 0.03 usr  0.00 sys +  8.62 cusr  0.94 csys =  9.59 CPU)
Result: PASS
(venv) kinow@ranma:~/Development/python/workspace/cylc-flow$ git log -n 1
commit b5f6ab92bbef5f935d40554ebf06b35952e7313c (HEAD -> pr-3581)
Author: Oliver Sanders <oliver.sanders@metoffice.gov.uk>
Date:   Fri Apr 24 15:03:33 2020 +0100
    tests: fix trigger-edit tests
```

**Comment by kinow** [comment]
Kicking Travis

**Comment by kinow** [comment]
Two approvals, and Travis is now happy. LGTM

**Pull request merged**